modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/BlockListener.java

```diff
@@ -67,6 +71,33 @@ public void onBlockBreak(@NotNull BlockBreakEvent e) {
         }
     }
 
+    //Prevents non pilots from placing blocks on your ship.
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onBlockPlace(BlockPlaceEvent e) {
+        if (!Settings.ProtectPilotedCrafts)
+            return;
+
+        Player p = e.getPlayer();
+
+        MovecraftLocation placedLocation = MathUtils.bukkit2MovecraftLoc(e.getBlockAgainst().getLocation());
+        for (Craft craft : CraftManager.getInstance().getCraftsInWorld(e.getBlock().getWorld())) {
+            if (craft == null || craft.getDisabled())
+                continue;
+
+            if(!(craft instanceof PlayerCraft)) {
+                continue;
+            }
+            if (((PlayerCraft) craft).getPilot() == p) {
+                continue;
+            }
```

Comment:
I believe that should be `PilotedCraft` not `PlayerCraft` to also protect squadron craft.

Reply:
This is now fixed I believe. If there is anything else, I would like this to be merged. On a side note, I was originally considering there to be a separate config option for this, but "protectPilotedCrafts" seems to be appropriate enough. What do you think?